![New York City schoolbus](schoolbus.jpg)

Photo by [Jannis Lucas](https://unsplash.com/@jannis_lucas) on [Unsplash](https://unsplash.com).
<br>

Every year, American high school students take SATs, which are standardized tests intended to measure literacy, numeracy, and writing skills. There are three sections - reading, math, and writing, each with a **maximum score of 800 points**. These tests are extremely important for students and colleges, as they play a pivotal role in the admissions process.

Analyzing the performance of schools is important for a variety of stakeholders, including policy and education professionals, researchers, government, and even parents considering which school their children should attend. 

I used the `schools.csv` file for my analysis that I found on datacamp, which is previewed below.

I have attempted to use my learnings about pandas and pivot tables to find answers to a few questions: 
1. What schools in NYC have the best math results?
2. What are the top 10 performing schools based on the combined SAT scores?
3. Which single borough has the largest standard deviation in the combined SAT score?

In [67]:
# Re-run this cell 
import pandas as pd

# Read in the data
schools = pd.read_csv("schools.csv")

# Preview the data
schools.head()


school_name  ... percent_tested
0  New Explorations into Science, Technology and ...  ...            NaN
1                               Essex Street Academy  ...           78.9
2                       Lower Manhattan Arts Academy  ...           65.1
3    High School for Dual Language and Asian Studies  ...           95.9
4      Henry Street School for International Studies  ...           59.7

[5 rows x 7 columns]

1. What schools in NYC have the best math results?

In [68]:
# Filtering the school dataframe
best_math = schools[schools["average_math"] >= (0.8 * 800)]

# Taking relevant columns and sorting
best_math_schools = best_math[["school_name", "average_math"]].sort_values("average_math", ascending = False)

# Viewing the dataframe
best_math_schools

school_name  average_math
88                              Stuyvesant High School           754
170                       Bronx High School of Science           714
93                 Staten Island Technical High School           711
365  Queens High School for the Sciences at York Co...           701
68   High School for Mathematics, Science, and Engi...           683
280                     Brooklyn Technical High School           682
333                        Townsend Harris High School           680
174  High School of American Studies at Lehman College           669
0    New Explorations into Science, Technology and ...           657
45                       Eleanor Roosevelt High School           641

There are 10 schools in NYC that have had an average math score in the SAT that is higher than 640/800 i.e., atleast 80% of total possible math score.

2. What are the top 10 performing schools based on the combined SAT scores?

In [69]:
# Making the total_SAT column
schools["total_SAT"] = schools["average_math"] + schools["average_reading"] + schools["average_writing"]

# Sorting to find top scoring schools and only required columns
schools_sorted = schools.sort_values("total_SAT", ascending = False)[["school_name", "total_SAT"]]

# Taking a list of only top 10 schools
top_10_schools = schools_sorted[0:10]

# Previewing the dataframe
top_10_schools

school_name  total_SAT
88                              Stuyvesant High School       2144
170                       Bronx High School of Science       2041
93                 Staten Island Technical High School       2041
174  High School of American Studies at Lehman College       2013
333                        Townsend Harris High School       1981
365  Queens High School for the Sciences at York Co...       1947
5                       Bard High School Early College       1914
280                     Brooklyn Technical High School       1896
45                       Eleanor Roosevelt High School       1889
68   High School for Mathematics, Science, and Engi...       1889

These are the top 10 schools in NYC with the highest average SAT scores, ranging from 2144 - 1889, i.e., 89.33% - 78.70%

3. Which single borough has the largest standard deviation in the combined SAT score?

In [70]:
# Grouping with respect to borough 
borough_data = schools.groupby("borough")["total_SAT"].agg(["count", "mean", "std"]).round(2)

# rename columns 
borough_data = borough_data.rename(columns={"count" : "num_schools", "mean": "average_SAT", "std": "std_SAT"})

# Displaying data of all 5 borough - 375 schools combined
borough_data

num_schools  average_SAT  std_SAT
borough                                         
Bronx                   98      1202.72   150.39
Brooklyn               109      1230.26   154.87
Manhattan               89      1340.13   230.29
Queens                  69      1345.48   195.25
Staten Island           10      1439.00   222.30

In [71]:
# Selecting the borough with the largest std deviation
largest_std_dev = borough_data[borough_data["std_SAT"] == borough_data["std_SAT"].max()]

# Displaying the data of the borough with the largest std dev
largest_std_dev.reset_index(inplace=True)
largest_std_dev

borough  num_schools  average_SAT  std_SAT
0  Manhattan           89      1340.13   230.29

The Borough with the largest std dev in SAT scores is thus Manhattan with a standard deviation of 230.29.